sockeye/arguments.py

```diff
@@ -1009,6 +1014,18 @@ def add_inference_args(params):
                                help='Beta factor for the length penalty used in beam search: '
                                     '(beta + len(Y))**alpha/(beta + 1)**alpha. Default: %(default)s')
 
+    decode_params.add_argument('--encoder-dtype',
```

Comment:
same comments as above, but more generally: is there a reason to allow different dtypes for encoder/ decoder?
If so I'd suggest going for the 'two int' flag style we have for other arguments as well (see for example `--num-embed`)

Reply:
Spoke with Tobi about this.  This is present only for debugging purposes in this version.  In future versions we propose having a single dtype.  This dtype should actually not need to be set during inference as it should be present in the model config.  Setting it will be enabled for cases where the user wants to override behaviour (for example converting from float32 params in an existing model to float16).